prow/scripts/provision-vm-and-start-reconciler-component-integration-k3d.sh

```diff
@@ -0,0 +1,125 @@
+  log::info "Execute Job Guard"
+  "${TEST_INFRA_SOURCES_DIR}/development/jobguard/scripts/run.sh"
+fi
+
+cleanup() {
+  # TODO - collect junit results
+  log::info "Stopping instance reconciler-component-integration-test-${RANDOM_ID}"
+  log::info "It will be removed automatically by cleaner job"
+
+  # do not fail the job regardless of the vm deletion result
+  set +e
+
+  gcloud compute instances stop --async --zone="${ZONE}" "reconciler-component-integration-test-${RANDOM_ID}"
+
+  log::info "End of cleanup"
+}
+
+function testCustomImage() {
+  CUSTOM_IMAGE="$1"
+  IMAGE_EXISTS=$(gcloud compute images list --filter "name:${CUSTOM_IMAGE}" | tail -n +2 | awk '{print $1}')
+  if [[ -z "$IMAGE_EXISTS" ]]; then
+    log::error "${CUSTOM_IMAGE} is invalid, it is not available in GCP images list, the script will terminate ..." && exit 1
+  fi
+}
+
+gcp::authenticate \
+  -c "${GOOGLE_APPLICATION_CREDENTIALS}"
+
+RANDOM_ID=$(openssl rand -hex 4)
+
+LABELS=""
+if [[ -z "${PULL_NUMBER}" ]]; then
+  LABELS=(--labels "branch=$PULL_BASE_REF,job-name=reconciler-integration")
+else
+  LABELS=(--labels "pull-number=$PULL_NUMBER,job-name=reconciler-integration")
+fi
+
+POSITIONAL=()
+while [[ $# -gt 0 ]]; do
+  key="$1"
+
+  case ${key} in
+    --image)
+      IMAGE="$2"
+      testCustomImage "${IMAGE}"
+      shift
+      shift
+      ;;
+    --*)
+      echo "Unknown flag ${1}"
+      exit 1
+      ;;
+    *)    # unknown option
+      POSITIONAL+=("$1") # save it in an array for later
+      shift # past argument
+      ;;
+  esac
+done
+set -- "${POSITIONAL[@]}" # restore positional parameters
+
+
+if [[ -z "$IMAGE" ]]; then
+  log::info "Provisioning vm using the latest default custom image ..."
+
+  IMAGE=$(gcloud compute images list --sort-by "~creationTimestamp" \
+       --filter "family:custom images AND labels.default:yes" --limit=1 | tail -n +2 | awk '{print $1}')
+
+  if [[ -z "$IMAGE" ]]; then
+    log::error "There are no default custom images, the script will exit ..." && exit 1
+  fi
+fi
+
+ZONE_LIMIT=${ZONE_LIMIT:-5}
+EU_ZONES=$(gcloud compute zones list --filter="name~europe" --limit="${ZONE_LIMIT}" | tail -n +2 | awk '{print $1}')
+STARTTIME=$(date +%s)
+for ZONE in ${EU_ZONES}; do
+  log::info "Attempting to create a new instance named reconciler-component-integration-test-${RANDOM_ID} in zone ${ZONE} using image ${IMAGE}"
+  gcloud compute instances create "reconciler-component-integration-test-${RANDOM_ID}" \
+      --metadata enable-oslogin=TRUE \
+      --image "${IMAGE}" \
+      --machine-type n2-standard-4 \
+      --zone "${ZONE}" \
+      --boot-disk-size 200 "${LABELS[@]}" && \
+  log::info "Created reconciler-component-integration-test-${RANDOM_ID} in zone ${ZONE}" && break
+  log::error "Could not create machine in zone ${ZONE}"
+done || exit 1
+ENDTIME=$(date +%s)
+echo "VM creation time: $((ENDTIME - STARTTIME)) seconds."
+
+trap cleanup exit INT
+
+log::info "Preparing environment variables for the instance"
+envVars=(
```

Comment:
this env shouldn't be hardcoded. think about istio tests or other reconciler tests

Reply:
we are just passing env from prowjob config to the provisioned VM